platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,373 @@
+- Checking all HTML pages for broken links
+
+The script can be found [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/index.js).
+
+#### CI pipeline
+The CI pipeline for `vets-website` triggers the above build and runs a number of
+tests. Pertinent to this document are:
+- Accessibility tests on static content pages
+  - Using the [`npm run nightwatch:docker --
+    --env=accessibility`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile#L82) command
+  - This iterates through all the pages on the sitemap, which is created by the
+    content build script
+- Broken link checks on static content pages
+  - The actual link checking happens during the build script, but it isn't
+    reported anywhere until [after the build script has been run](https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L200)
+    
+#### Deployment
+There are currently two kinds of deployments:
+1. [Partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#partial-deploy--static-page-changes-only)
+    - For static pages only
+    - Uses the latest `vets-website` release and static assets
+      - Does not trigger Webpack to build the JS and CSS bundles
+      - Instead,
+        [fetches](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/configure-assets.js#L22-L23)
+        the assets [from S3](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/download-assets.js#L111)
+        - Process is triggered [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile.content#L32)
+    - Does **NOT** run any content validation such as accessibility tests or
+      broken link checker
+2. [Full deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#full-deploy-of-vagov-client-app)
+    - For `vets-website` code changes
+    - Creates a new release and deploys it
+    - Fetches the latest Drupal content
+      - Falls back to the cached content in S3 from the last successful deploy
+        (full or partial)
+        <!-- TODO: Verify that either deploy will cache the pages.json and files -->
+    - Runs the accessibility tests and link checker
+
+See [here](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/)
+for more details.
+
+#### Pain points
+- Engineers encounter **confusing build errors** that are due to the mismatch in
+  outdated locally cached content and updated Liquid templates
+- Fetching content from the CMS and building HTML files when developing
+  applications **slows down the local builds** for engineers unnecessarily
+- When their applications have nothing to with the CMS
```

Comment:
is this an unfinished sentence, or a misplaced bullet point?

Reply:
Ah, dang it. Yeah, I rephrased this section and that point got orphaned. :face_with_head_bandage: